### DISCUSSION TITLE
Using hunger for flight?

### DISCUSSION THREAD

**Discussion by DaringShepard** [Zrips/CMI#2298] [Feature Requests]
Could we get a feature where flying drains hunger? Similar to the Flight Charge system, but instead of draining the flight charge, it drains from your hunger instead.

### Detailed explanation of what it should do, when and in what situation: ###
Users with a permission, let's just say `cmi.hunger.flightdrain`, should have their hunger drained while flying, and when they run out of food, they loose the ability to fly. The drain amount/rate should also probably be configurable, and being able to set this with a permission would be fantastic.

### Basic explanation as to why this feature is needed and/or useful to have: ###
I'd say this would be pretty useful to have since a plugin like this doesn't exist past 1.12, aside from LiteFly, which requires use of an item. Plus, having an alternative to the Flight Charge, aside from just giving flight for free, would be a nice addition.

**Comment by Zrips** [comment]
Its harder to work with hunger value as its only goes (in normal conditions) up to 20. Which is quite small number to be used for actual flight. Some allot more complex tracking should be added to keep full track of  hunger while keeping track of possible hunger value in more accurate state, like 15.23 of hunger. Idea is good, but i need to think some better approach how to handle it
